src/main/java/de/taimos/pipeline/aws/S3PresignUrlStep.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * -
+ * #%L
+ * Pipeline: AWS Steps
+ * %%
+ * Copyright (C) 2016 Taimos GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package de.taimos.pipeline.aws;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.google.common.base.Preconditions;
+import de.taimos.pipeline.aws.utils.StepUtils;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.TaskListener;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.joda.time.DateTime;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+import java.net.URL;
+import java.util.Date;
+import java.util.Set;
+
+public class S3PresignUrlStep extends AbstractS3Step {
+
+	private final String bucket;
+	private final String key;
+
+	@DataBoundConstructor
+	public S3PresignUrlStep(String bucket, String key, boolean pathStyleAccessEnabled, boolean payloadSigningEnabled) {
+		super(pathStyleAccessEnabled, payloadSigningEnabled);
+		this.bucket = bucket;
+		this.key = key;
+	}
+
+	public String getBucket() {
+		return bucket;
+	}
+
+	public String getKey() {
+		return key;
+	}
+
+	@Override
+	public StepExecution start(StepContext context) throws Exception {
+		return new S3PresignUrlStep.Execution(this, context);
+	}
+
+	@Extension
+	public static class DescriptorImpl extends StepDescriptor {
+
+		@Override
+		public Set<? extends Class<?>> getRequiredContext() {
+			return StepUtils.requires(TaskListener.class, EnvVars.class, FilePath.class);
+		}
+
+		@Override
+		public String getFunctionName() {
+			return "s3PresignURL";
+		}
+
+		@Override
+		public String getDisplayName() {
+			return "Presign file in S3";
+		}
+	}
+
+	public static class Execution extends StepExecution {
```

Comment:
Please use a `SynchronousNonBlockingStepExecution` here to make the code easier.

Reply:
Cool. didn't know about that implementation